src/Microsoft.AspNetCore.Mvc.Razor.Extensions/PageDirective.cs

```diff
@@ -69,13 +75,25 @@ private static string TrimQuotes(string content)
 
         private class Visitor : IntermediateNodeWalker
         {
-            public DirectiveIntermediateNode DirectiveNode { get; private set; }
+            public IntermediateNode DirectiveNode { get; private set; }
+
+            public IEnumerable<DirectiveTokenIntermediateNode> DirectiveTokens { get; private set; }
```

Comment:
The node is an `IntermediateNode` because it can be a malformed node or a non-malformed node.